rfcs/0000-multiple-kuadrant-instances.md

```diff
@@ -0,0 +1,152 @@
+
+![](https://i.imgur.com/yChVsT6.png)
+
+### All the gateways referenced by a single policy must belong to the same kuadrant instance
+
+The Gateway API allows, in its latest version  [v1beta1](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.CommonRouteSpec), an HTTPRoute to have multiple gateway parents. Thus, a kuadrant policy might technically target multiple gateways managed by multiple kuadrant instances. Kuadrant does **not** support this use case.
+
+![](https://i.imgur.com/ZpsBf4i.png)
+
+The main reason is related to the rate limiting capability. The limits specified in the RateLimit Policy would be enforced per kuadrant instance basis (provided by Limitador instance). Thus, traffic hitting one gateway would see different rate limiting counters than traffic hitting the other gateway. The user would expect X rps and actually it would be X rps per gateway. For consistency reasons, when this configuration happens, the control plane will reject the policy.
+
+### The Kuadrant CRD
+
+Currently, the Kuadrant CRD has an empty spec.
+
+```yaml
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant-sample
+spec: {}
+```
+
+According to the definition above of the kuadrant instance,
+the proposed new Kuadrant CRD would add a label __selector__ to specify which gateways that instance would manage.
+
+```yaml
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant-a
+spec:
+  gatewaysSelector:
+    matchLabels:
+      app: kuadrant-a
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Wiring Kuadrant policies with Kuadrant instances
+Technically, the Kuadrant policies do not belong to any Kuadrant instance. At any moment of time, one policy can switch the targeted network resource specified in the `spec` from one gateway to another. Directly or indirectly via the HTTPRoute. The target references are dynamic by nature, so is the list of gateways to which kuadrant policies should apply.
+Thus, the Kuadrant's control plane needs a procedure to associate a policy with **one** kuadrant instance at any time. When the control plane knows which kuadrant instance is affected, the policy rules can be used to configure the Limitador and Authorino instances belonging to that kuadrant's instance. Since the associated kuadrant instance of a policy is dynamic by nature, this procedure must be executed on  every event related to the policy.
+
+When the policy's `targetRef` targets a Gateway, there is a direct reference to the gateway.
+
+When the policy's `targetRef` targets an HTTPRoute, Kuadrant will follow the [`parentRef`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.CommonRouteSpec) attribute which should be a direct reference to the gateway or gateways.
```

Comment:
This means that indirectly, a KAP or RLP, would be applied to more than 1 Kuadrant instance. As mentioned before, the control plane in charge of associating policy with 1 Kuadrant instance, will need to decide which policy apply in case there's another HTTPRoute sharing one or more of `parentRef` gateways

Reply:
OK, it's mentioned that this use case is [not supported](https://github.com/Kuadrant/architecture/pull/5/files#diff-416c18ad603a7ea22bfb07cf5c9434220f70097b4746c60e4bb1b527496467bdR51) , still we might need to provide a way to mitigate since one might not have access to modify the network topology